aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1701,6 +1749,41 @@ impl VMAdapter for AptosVM {
             },
         })
     }
+
+    fn execute_single_transaction_sequential(
+        &self,
+        txn: &PreprocessedTransaction,
+        view: &impl StateView,
+        log_context: &AdapterLogSchema,
+    ) -> Result<(VMStatus, VMOutput, Option<String>), VMStatus> {
+        match self.execute_single_transaction(txn, &view.as_move_resolver(), log_context, true) {
+            Ok((vm_status, vm_output, sender)) => match vm_output.try_materialize(view) {
+                Ok(vm_output) => {
+                    return Ok((vm_status, vm_output, sender));
+                },
+                Err(vm_status) => {
+                    if !vm_status.is_aggregator_error() {
+                        return Err(vm_status);
+                    }
+                },
+            },
+            Err(vm_status) => {
+                if !vm_status.is_aggregator_error() {
+                    return Err(vm_status);
+                }
+            },
+        }
+        match self.execute_single_transaction(txn, &view.as_move_resolver(), log_context, false) {
```

Comment:
What do you think about removing this match completely and instead having an assert in `execute_single_transaction`:
```
PreprocessedTransaction::UserTransaction(txn) => {
  ...
  let (vm_status, output) =
                    self.execute_user_transaction(resolver, txn, log_context, aggregator_enabled);
  assert!(aggregator_enabled || output.delta_change_set().is_empty());
  ...

```

Reply:
Done.